File: Python/559.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def maxDepth(self, root: 'Node') -> int:
        if not root: return 0
        return self.getHeight(root, 1)


    def getHeight(self, root: 'Node', currentHeight: int) -> int:
        if not root.children: return currentHeight
        else:
            heightList = []
            for child in root.children:
                heightList.append(self.getHeight(child, currentHeight + 1))

        return max(heightList)

            

File: CPP/559.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    int maxDep = 0;
    int maxDepth(Node* root) {
        if (!root) return 0;
        getHeight(root, 1);
        return maxDep;
    }

    void getHeight(Node* root, int currentHeight){
        if (!root) return;
        // Update current level detph
        maxDep = max(maxDep, currentHeight);

        for (auto child : root->children){
            getHeight(child, currentHeight + 1);
        }
    }
};
